upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -1088,6 +1088,25 @@ func (b *BootstrapChannelBuilder) buildAddons() *channelsapi.Addons {
 					Id:                id,
 				})
 			}
+		} else if kops.CloudProviderID(b.Cluster.Spec.CloudProvider) == kops.CloudProviderAWS {
+			key := "aws-cloud-controller.addons.k8s.io"
+			version := "v1.18.0-alpha.1"
+
+			location := key + "/v1.18.0.yaml"
+			id := "k8s-1.18-ccm"
+
+			addons.Spec.Addons = append(addons.Spec.Addons, &channelsapi.AddonSpec{
+				Name:              fi.String(key),
+				Version:           fi.String(version),
+				Manifest:          fi.String(location),
+				Selector:          map[string]string{"k8s-addon": key},
+				KubernetesVersion: ">=1.18.0",
+				Id:                id,
+			})
+
+			// Ensure ExternalCloudVolumePlugin is set to AWS
+			b.Cluster.Spec.KubeControllerManager.ExternalCloudVolumePlugin = "aws"
```

Comment:
This type of logic can be added in `./pkg/model/components`. That way changes done are also subject to validation and such. There is already something similar in place for openstack.

Reply:
Thanks, that's very helpful.  Will add some validation here.